dashboard/app/models/sections/section.rb

```diff
@@ -237,6 +237,16 @@ def summarize
       link_to_assigned = script_path(script)
     end
 
+    # Some scripts are associated with a course (e.g. csp1-2018 is the script
+    # for "CSP Unit 1 - The Internet ('18-'19)", which is part of the csp18-19 # course. Courses have different versions based on year; similar courses
+    # across years have a family_name (either CSD or CSP). We want to pass the # family_name associated with a script, if there is one, so that we can
+    # determine whether to show the sharing column on the Manage Students Table # of Teacher Dashboard.
+    if script.try(:courses)
+      if script.courses[0] && script.courses[0].properties
+        family_name = script.courses[0].properties["family_name"]
+      end
+    end
+
```

Comment:
@davidsbailey What do you think about this approach? It solves for the problem of having to add script names every year. I need to make sure it works when:

- [ ] no script, no course assigned 
- [ ] course is assigned 
- [ ] script in a course is assigned 
- [ ] script not in a course is assigned 

I'm using  `script.courses`; should I be concerned about times when a script is associated with more than one course?  Does that happen or are scripts always specific to one course?

Reply:
nice solution @Erin007 ! Seems to be version-proof. One naming request though: `section.script.family_name` is misleading because scripts also have family names, e.g. https://github.com/code-dot-org/code-dot-org/blob/74c2f8589d1041d4743e05182ad1e88389864df4/dashboard/config/scripts/csd1-2017.script#L10

If you don't want to refactor this to be `section.course.family_name`, then you could do a name-only change to call this `section.script.course_family_name`. How does that sound?

Reply:
looking closer, I think `section.course.family_name` will be plenty easy, so that's what I'd recommend unless you see a reason not to.